tiledb/sm/array_schema/array_schema.cc

```diff
@@ -580,8 +580,13 @@ Status ArraySchema::deserialize(ConstBuffer* buff) {
     RETURN_NOT_OK(cell_validity_filters_.deserialize(buff));
 
   // Load domain
+  auto&& [st_domain, deserialized_domain]{Domain::deserialize(buff, version_)};
```

Comment:
This call is missing arguments for cell and tile order. They need to be initialized from the values in the array schema. See below for more on the arguments to `deserialize`.